### ISSUE TITLE
Arbitrary font size preference is allowed

### ISSUE THREAD

**Issue by johnblommers** [marktext/marktext#264]
### Description

If you set a large font size in the preferences marktext does not complain and evidently complies. As a a result it is impossible to see anything in the application window.

### Steps to reproduce

1. Change the font size to 2416 in the preferences
2. Save the preferences and exit marktext
3. Open marktext and try to edit the preferences. You get white window contents.

**Expected behavior:**
marktext notices the font size is crazy large, complains to the user and replaces it with a default of 16 pts.

**Actual behavior:** 

All you see is a blank page. No error message is displayed

**Example preference file:** 

```
{
  "fontSize": "2416px",
  "lightColor": "rgb(48, 49, 51)",
  "darkColor": "rgb(217, 217, 217)",
  "lineHeight": 2,
  "theme": "light",
  "autoSave": true,
  "aidou": false,
  "preferLooseListItem": true,
  "autoPairBracket": true,
  "autoPairMarkdownSyntax": true,
  "autoPairQuote": true
}
```

### Versions

- Mark Text: v 0.10.21
- OS: macOS 10.11.6 El Capitan

**Comment by Jocs** [comment]
@johnblommers 

We are also considering whether to modify the current configuration file into a visual configuration interface so that similar problems can be avoided.
 
The configuration file is more friendly to the developer. The visual configuration interface can do some verification processing on the input. Which one do you prefer?

**Comment by johnblommers** [comment]
The visual configuration interface is far more user-friendly and I always put the user first.

**Comment by fxha** [comment]
Some  key points why we should build a GUI for settings:

- easier to use from a users point of view
- less error prone (because someone type in invalid values)
- we can integrate custom themes and plugin-settings later
- we do not have to convert old settings to new settings (and support old settings format) - because of pre-release version
- @Jocs we can seperate settings from GUI (Mark Text) and the "core"

Furthermore we should add a `--safe` mode in the new settings to run Mark Text without loading user settings, themes and plugins. So that we/the user can test issues in "safe mode" without loading user configuration.

**Cons:**

- more work for us 😄 
- we delete all user settings

**Comment by Jocs** [comment]
>Furthermore we should add a --safe mode in the new settings to run Mark Text without loading user settings, themes and plugins. So that we/the user can test issues in "safe mode" without loading user configuration.

How about a `reset to default setting` ?

**Comment by fxha** [comment]
> How about a reset to default setting ?

Sure also a good idea, but I don't want to reset my settings to test an issue.

**Comment by Jocs** [comment]
Related with #324

**Comment by fxha** [comment]
Resolved in PR #1003
